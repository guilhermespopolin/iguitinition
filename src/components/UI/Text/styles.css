.text { margin: 1rem 0; }

.text-size--small { font-size: var(--font-small); }

.text-size--medium { font-size: var(--font-medium); }

.text-size--large { font-size: var(--font-large); }

.text-alignment--left { text-align: left; }

.text-alignment--center { text-align: center; }

.text-alignment--right { text-align: right; }

.text-alignment--justify { text-align: justify; }