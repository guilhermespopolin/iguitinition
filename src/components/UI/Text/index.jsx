import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import styles from './styles.css'

function Text({
  className,
  children,
  size,
  alignment,
  ...others
}) {
  return (
    <p
      className={cx(
        styles.text,
        styles[`text-size--${size}`],
        styles[`text-alignment--${alignment}`],
        className,
      )}
      {...others}
    >
      {children}
    </p>
  )
}

Text.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  alignment: PropTypes.oneOf(['left', 'center', 'right', 'justify']),
}

Text.defaultProps = {
  className: '',
  size: 'medium',
  alignment: 'justify',
}

export default Text
