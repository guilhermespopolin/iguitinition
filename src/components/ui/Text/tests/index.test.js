import React from 'react'
import { cleanup } from 'react-testing-library'
import renderComponent from '__tests__/utils'

import Text from '..'

afterEach(cleanup)

describe('<Text />', () => {
  describe('Based on size modifier', () => {
    describe('When size modifier equals to `small`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text size="small">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })

    describe('When size modifier equals to `medium`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text size="medium">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })

    describe('When size modifier equals to `large`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text size="large">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })
  })

  describe('Based on alignment modifier', () => {
    describe('When alignment modifier equals `left`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text alignment="left">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })

    describe('When alignment modifier equals `center`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text alignment="center">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })

    describe('When alignment modifier equals `right`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text alignment="right">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })

    describe('When alignment modifier equals `justify`', () => {
      it('should render properly', () => {
        const { getByText } = renderComponent(<Text alignment="justify">Sample Text</Text>)
        const textNode = getByText('Sample Text')

        expect(textNode).toMatchSnapshot()
      })
    })
  })
})
