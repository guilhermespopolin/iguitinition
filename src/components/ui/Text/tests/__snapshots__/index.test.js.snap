// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Based on alignment modifier When alignment modifier equals \`center\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 1.2rem;
  text-align: center;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on alignment modifier When alignment modifier equals \`justify\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 1.2rem;
  text-align: justify;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on alignment modifier When alignment modifier equals \`left\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 1.2rem;
  text-align: left;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on alignment modifier When alignment modifier equals \`right\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 1.2rem;
  text-align: right;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on size modifier When size modifier equals to \`large\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 2.5rem;
  text-align: justify;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on size modifier When size modifier equals to \`medium\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 1.2rem;
  text-align: justify;
}

<p
  class="c0"
>
  Sample Text
</p>
`;

exports[`<Text /> Based on size modifier When size modifier equals to \`small\` should render properly 1`] = `
.c0 {
  margin: 1rem 0;
  font-size: 0.9rem;
  text-align: justify;
}

<p
  class="c0"
>
  Sample Text
</p>
`;
